### PR TITLE
[IoT] `az iot dps`: Add support for managed identities within DPS

### PR DESCRIPTION
**Pull request by vilit1**
**Related command**
`az iot dps create`
`az iot dps identity`

**Description**

Addition of system and user assigned identity to DPS with the newer API version. The required sdk has been updated to point to the newer sdk.

**Testing Guide**
Created a new test - test_iot_dps_commands.py::IoTDpsTest::test_dps_identity_lifecycle

**History Notes**
[IoT] `az iot dps`: Add support for managed identities within DPS

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
IoT

**Comment by yonzhan** [comment]
Please fix CI issues first.

**Comment by zhoxing-ms** [comment]
@vilit1 Could you please resolve these CI issues?

**Comment by zhoxing-ms** [comment]
Please refer to the [guideline](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests) to write the description of those PR changes in the `History Notes` section of PR description. Please note that it will be synced to the [release notes](https://learn.microsoft.com/en-us/cli/azure/release-notes-azure-cli?toc=%2Fcli%2Fazure%2Ftoc.json&bc=%2Fcli%2Fazure%2Fbreadcrumb%2Ftoc.json) of CLI. Such as #26261 
image

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 1]
```diff
@@ -101,6 +101,12 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.custom_command('verify', 'iot_dps_certificate_verify')
         g.custom_command('update', 'iot_dps_certificate_update')
 
+    # iot dps identity commands
+    with self.command_group('iot dps identity', client_factory=iot_service_provisioning_factory, is_preview=True) as g:
+        g.custom_command('assign', 'iot_dps_identity_assign')
+        g.custom_show_command('show', 'iot_dps_identity_show')
+        g.custom_command('remove', 'iot_dps_identity_remove')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:108]: Do we need to consider adding `confirmation=true` for the remove command?

**Comment by zhoxing-ms** [comment]
@vilit1 Please note that we are launching the release for this sprint this week. Please resolve all comments by tomorrow, otherwise the release of this PR will have to be postponed to the next sprint (on 08-01)

**Comment by vilit1** [comment]
> @vilit1 Please note that we are launching the release for this sprint this week. Please resolve all comments by tomorrow, otherwise the release of this PR will have to be postponed to the next sprint (on 08-01)

This is ok. I can wait till the next release.

File: src/azure-cli/azure/cli/command_modules/iot/commands.py (hunk 1 above)
Comment (vilit1) [reply src/azure-cli/azure/cli/command_modules/iot/commands.py:108]: There is no need, a command that is similar to it `az iot hub identity remove` does not have the confirmation.

**Pull request closed without merging**